File: chp2/python/CurrentConditionsDisplay.py
from Observer import Observer
from DisplayElement import DisplayElement


class CurrentConditionsDisplay(Observer, DisplayElement):
    def __init__(self, weatherData):
        super(CurrentConditionsDisplay, self).__init__()
        self.weatherData = weatherData
        self.weatherData.registerObserver(self)

    def update(self, temperature, humidity, pressure):
        self.temperature = temperature
        self.humidity = humidity
        self.display()

    def display(self):
        print "Current conditions: %sF degrees and %s%% humidity" % (self.temperature, self.humidity)

File: chp2/java/src/CurrentConditionsDisplay.java
public class CurrentConditionsDisplay implements Observer, DisplayElement {
    private float tempterature;
    private float humidity;
    private Subject weatherData;

    public CurrentConditionsDisplay(Subject weatherData) {
	this.weatherData = weatherData;
	weatherData.registerObserver(this);
    }

    public void update(float tempterature, float humidity, float pressure) {
	this.tempterature = tempterature;
	this.humidity = humidity;
	display();
    }

    public void display() {
	System.out.println("Current conditions: " + tempterature + "F degrees and " + humidity + "% humidity");
    }
}
